File: C++/10845.cpp
#include<iostream>
#include<queue>
using namespace std;

int main() {
	int n;
	string cmd;
	int param;
	queue<int> q;
	cin >> n;
	while (n--) {
		cin >> cmd;
		if (cmd == "push") {
			cin >> param;
			q.push(param);
		}
		else if (cmd == "pop") {
			if (q.empty()) {
				cout << -1 << endl;
			}
			else {
				cout << q.front() << endl;
				q.pop();
			}
		}
		else if (cmd == "size") {
			cout << q.size() << endl;
		}
		else if (cmd == "empty") {
			cout << q.empty() << endl;
		}
		else if (cmd == "front") {
			if (q.empty()) {
				cout << -1 << endl;
			}
			else {
				cout << q.front() << endl;
			}
		}
		else if (cmd == "back") {
			if (q.empty()) {
				cout << -1 << endl;
			}
			else {
				cout << q.back() << endl;
			}
		}
	}
}

File: Python/10845.py
import sys

N = int(sys.stdin.readline())
q = []

for x in range(N):
    comnd = sys.stdin.readline().rstrip()
    # print('cmond:', comnd)

    if comnd == 'pop':
        if q:
            print(q.pop(0))
        else:
            print(-1)
    elif comnd == 'size':
        print(len(q))
    elif comnd == 'empty':
        if q:
            print(0)
        else:
            print(1)
    elif comnd == 'front':
        if q:
            print(q[0])
        else:
            print(-1)
    elif comnd == 'back':
        if q:
            print(q[-1])
        else:
            print(-1)
    else:
        q.append(comnd.split()[-1])
    # print('q:', q)
